test/calibration/experiments/test_rabi.py

```diff
@@ -0,0 +1,129 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Rabi amplitude Experiment class."""
+
+from typing import Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit, execute
+from qiskit.providers.basicaer import QasmSimulatorPy
+from qiskit.test import QiskitTestCase
+from qiskit.qobj.utils import MeasLevel
+
+from qiskit_experiments import ExperimentData
+from qiskit_experiments.calibration.experiments.rabi import RabiAnalysis, Rabi
+from qiskit_experiments.data_processing.data_processor import DataProcessor
+from qiskit_experiments.data_processing.nodes import Probability
+from qiskit_experiments.test.mock_iq_backend import IQTestBackend
+
+
+class RabiBackend(IQTestBackend):
+    """A simple and primitive backend, to be run by the T1 tests."""
+
+    def __init__(
+        self,
+        iq_cluster_centers: Tuple[float, float, float, float] = (1.0, 1.0, -1.0, -1.0),
+        iq_cluster_width: float = 1.0,
+        amplitude_to_angle=np.pi,
+    ):
+        """Initialize the rabi backend."""
+        self._amplitude_to_angle = amplitude_to_angle
+
+        super().__init__(iq_cluster_centers, iq_cluster_width)
+
+    def _compute_probability(self, circuit: QuantumCircuit) -> float:
+        """Returns the probability based on the rotation angle and amplitude_to_angle."""
+        amp = next(iter(circuit.calibrations["rx"].keys()))[1][0]
+        return np.sin(self._amplitude_to_angle * amp) ** 2
+
+
+class TestRabiEndToEnd(QiskitTestCase):
+    """Test the rabi experiment."""
+
+    def test_rabi_end_to_end(self):
+        """Test the Rabi experiment end to end."""
+
+        backend = RabiBackend()
+
+        rabi = Rabi(3, np.linspace(-0.95, 0.95, 21))
+        result = rabi.run(backend).analysis_result(0)
+
+        self.assertEqual(result["quality"], "computer_good")
```

Comment:
how we can fit parameters?

Reply:
We can be a bit more precise. See here 217a27b . Important is to see that this picks up the variation in the frequency of the Rabi.